pkg/controllers/scheduling/schedule.go

```diff
@@ -113,13 +114,22 @@ type pluginScheduler struct {
 }
 
 func NewPluginScheduler(handle plugins.Handle) *pluginScheduler {
+	resourceAllocatableCPU := &resource.ResourceAllocatableCPU{}
```

Comment:
I think we can make the code more general here. For example we could have a ResourcePrioritizerFactory. So you will do like

```
resourcePrioritizerFactory := resource.NewPrioritizerFactory(handle)

resourcePrioritizerFactory.Resource("cpu").Mode("Allocatable").Build()
resourcePrioritizerFactory.Resource("memory").Mode("Allocatable").Build()
```
And you would not need to create 4 plugins, but a more general one. WDYT?

Reply:
Or we might want a dynamic generated prioritizer lists.

```go
prioritizerFactory := prioritizer.NewFactory(handle)
...

// In scheduler we build prioritizers for each placement
prioritizers := priotizerFactory.WithPlacement(placement).Build()
```
We can make it as a followup.